File: Source/Aladdin.CAPI.ANSI.PKCS11.Java/src/aladdin/capi/ansi/pkcs11/sign/rsa/VerifyData.java
package aladdin.capi.ansi.pkcs11.sign.rsa;
import aladdin.capi.*; 
import aladdin.capi.pkcs11.*; 

///////////////////////////////////////////////////////////////////////////
// Алгоритм проверки подписи RSA PKCS1
///////////////////////////////////////////////////////////////////////////
public class VerifyData extends aladdin.capi.pkcs11.VerifyData
{
    // конструктор
	public VerifyData(Applet applet, long algID) 
    
        // сохранить переданные параметры
        { super(applet); this.algID = algID; } private final long algID;

	// параметры алгоритма
    @Override 
    protected Mechanism getParameters(Session sesssion, IParameters parameters)
	{
		// параметры алгоритма
		return new Mechanism(algID); 
	}
}

File: Source/Aladdin.CAPI.ANSI.PKCS11/Sign/RSA/VerifyData.cs
using System;
using Aladdin.PKCS11; 

namespace Aladdin.CAPI.ANSI.PKCS11.Sign.RSA
{
    ///////////////////////////////////////////////////////////////////////////
    // Алгоритм проверки подписи RSA PKCS1
    ///////////////////////////////////////////////////////////////////////////
    public class VerifyData : CAPI.PKCS11.VerifyData
    {
        // конструктор
	    public VerifyData(CAPI.PKCS11.Applet applet, ulong algID) 
    
            // сохранить переданные параметры
            : base(applet) { this.algID = algID; } private ulong algID;

	    // параметры алгоритма
        protected override Mechanism GetParameters(
            CAPI.PKCS11.Session sesssion, IParameters parameters)
	    {
		    // параметры алгоритма
		    return new Mechanism(algID); 
	    }
    }
}
